lib/ansible/module_utils/vmware.py

```diff
@@ -639,3 +640,78 @@ def find_host_by_cluster_datacenter(module, content, datacenter_name, cluster_na
             return host, cluster
 
     return None, cluster
+
+
+def set_vm_power_state(content, vm, state, force):
+    """
+    Set the power status for a VM determined by the current and
+    requested states. force is forceful
+    """
+    facts = gather_vm_facts(content, vm)
+    expected_state = state.replace('_', '').lower()
+    current_state = facts['hw_power_status'].lower()
+    result = dict(
+        changed=False,
+        failed=False,
+    )
+
+    # Need Force
+    if not force and current_state not in ['poweredon', 'poweredoff']:
```

Comment:
Why do we need to force to change powerstate?  Seems over protective.

Reply:
It's optional, and required by the **vmware_guest** module.